array_api_tests/hypothesis_helpers.py

```diff
@@ -121,8 +125,17 @@ def two_broadcastable_shapes(draw, shapes=shapes):
 
 nonbroadcastable_ones_array_two_args = hypotheses_tuples(ones_arrays, ones_arrays)
 
-# TODO: Generate general arrays here, rather than just scalars.
-numeric_arrays = builds(full, just((1,)), floats())
+floating_arrays = xps.arrays(dtype=xps.floating_dtypes(), shape=xps.array_shapes())
+
+@composite
+def broadcastable_floating_array_pairs(draw):
+    dtype = draw(xps.floating_dtypes())
+    broadcastable_shapes = draw(xps.mutually_broadcastable_shapes(2, min_dims=1))
+    shape1, shape2 = broadcastable_shapes.input_shapes
+    assume(len(shape1) >= len(shape2))
```

Comment:
Why is this restriction here?

Reply:
Ahh this goes back to the issue I was having with the special case tests generating indices which (mostly) worked only when the first array had ndims `>=` the second array. This would be resolved if both arrays in the special case tests shared the same shape, as you suggested.

Reply:
I see. Yeah, there's no need to test broadcasting in the special cases tests (it's already tested separately in test_broadcasting.py), and it just makes building a mask more complicated, so we should just generate arrays of the same shape.